extensions/mapstruct/deployment/src/main/java/org/apache/camel/quarkus/component/mapstruct/deployment/MapStructProcessor.java

```diff
@@ -0,0 +1,372 @@
+                    ClassInfo mapperDefinitionClassInfo = index.getClassByName(mapperDefinitionClassName);
+                    Optional<FieldInfo> mapperInstanceField = mapperDefinitionClassInfo
+                            .fields()
+                            .stream()
+                            .filter(fieldInfo -> Modifier.isStatic(fieldInfo.flags()))
+                            .filter(fieldInfo -> fieldInfo.type().name().toString().equals(mapperDefinitionClassName))
+                            .findFirst();
+
+                    // Check of the Mapper is a CDI bean with one of the supported MapStruct annotations
+                    boolean mapperBeanExists = classInfo.hasDeclaredAnnotation(ApplicationScoped.class)
+                            || classInfo.hasDeclaredAnnotation(Named.class);
+                    if (mapperInstanceField.isEmpty()) {
+                        if (mapperBeanExists) {
+                            unremovableBean
+                                    .produce(new UnremovableBeanBuildItem(beanInfo -> beanInfo.hasType(classInfo.name())));
+                        } else {
+                            // Create the Mapper ourselves
+                            mapperRuntimeValue.set(recorder.createMapper(mapperClassName));
+                        }
+                    }
+
+                    /*
+                     * Generate SimpleTypeConverter.ConversionMethod implementations for each candidate Mapper method.
+                     *
+                     * ReflectionHelper is used to resolve the mapper methods for simplicity, compared to Jandex where
+                     * we potentially have to iterate over the type hierarchy (E.g for multiple interfaces,
+                     * interface / class inheritance etc).
+                     *
+                     * public final class FooConversionMethod implements ConversionMethod {
+                     *    private final CarMapperImpl mapper;
+                     *
+                     *    // Generated only if a Mapper instance is a CDI bean
+                     *    public ToCarConversionMethod() {
+                     *    }
+                     *
+                     *    // Generated only if a Mapper instance was declared on the Mapper interface
+                     *    public ToCarConversionMethod() {
+                     *        this(CarMapper.INSTANCE);
+                     *    }
+                     *
+                     *    public ToCarConversionMethod(CarMapperImpl mapper) {
+                     *        this.mapper = mapper;
+                     *    }
+                     *
+                     *    @Override
+                     *    public Object doConvert(Class<?> type, Exchange exchange, Object value) throws Exception {
+                     *        return mapper.stringToInt(value);
+                     *    }
+                     * }
+                     */
```

Comment:
:+1:  Nice doc

Reply:
Just noticed a few copy / paste errors that make it confusing (constructor method names) so I will correct that.